pkg/controller/volume/attachdetach/metrics/metrics_test.go

```diff
@@ -125,7 +126,11 @@ func TestVolumesInUseMetricCollection(t *testing.T) {
 		t.Errorf("Expected one volume in use got %d", len(nodeUseMap))
 	}
 	testNodeMetric := nodeUseMap["metric-test-host"]
-	pluginUseCount, ok := testNodeMetric["fake-plugin"]
+	fakePluginName := "fake-plugin"
```

Comment:
I wonder if we should split this into 2 tests. One for csi migration, and one for non-migrated plugins. For migration test, then we can explicitly turn on migration feature gates.

And for non-migration tests, we should use some other driver like CSI?

Ditto for all the tests below that are supposed to be driver-agnostic tests.

Reply:
Why is this test using GcePersistentDiskVolumeSource in the first place? Wouldn't it be better to just use the CSI host path driver instead? (I'm assuming something like HostPathVolumeSource wouldn't go through the attach-detatch controller and so not get metrics to test).

Reply:
For 1.23, I would suggest that we explicitly disable CSIMigrationGCE in these driver-agnostic tests, since the original intention was not for these tests to test migration.

Then in 1.24, we work on converting these tests to use a non-migratable volume type like CSI, and investigate if we want migration-specific tests.

Reply:
done

Reply:
Opened up https://github.com/kubernetes/kubernetes/issues/106439 to track future work